sequencer/api/light-client.toml

```diff
@@ -0,0 +1,43 @@
+[route.leaf]
+PATH = ["/leaf/:number", "/leaf/:number/:finalized"]
```

Comment:
I have a question about this, for integrations when we call this endpoint will it be like if I get a given hotshot block from query node, I will call this endpoint with that hotshot block which will give us this proof?

Reply:
If my understanding is right then we will also have to write some code similar to the below to verify this chain of leafs?

```
pub async fn verify(
        &self,
        quorum: &impl Quorum,
        finalized: Option<&Leaf2>,
    ) -> Result<LeafQueryData<SeqTypes>> {
```

Reply:
This will all be abstracted. The integrations will just call a light client function that has an interface identical to the current query service (so like `/availability/block/:height/namespace/:id` endpoint). But as a function instead of an HTTP call. Internally the light client will use this new query service endpoint as needed to both fetch and verify data.

> If my understanding is right then we will also have to write some code similar to the below to verify this chain of leafs?

Not sure what you mean, this function already exists as of this PR?